server/controllers/auth/login.js

```diff
@@ -1,3 +1,33 @@
-exports.login = (req, res) => {
-  res.send(' Hello from login file');
+const { compare } = require('bcrypt');
+const { jwtSign } = require('../../helpers');
+const { getUserByUsername } = require('../../database/queries/users');
+
+exports.login = (req, res, next) => {
+  const { username, password } = req.body;
+  if (!username || !password) next({ code: 400, msg: 'bad request !!!!' });
+
+  const key = process.env.KEY;
+  let id;
+  getUserByUsername(username)
+    .then(({ rows: [{ id: dbId, password: dbPassword }] }) => {
+      id = dbId;
+      return compare(password, dbPassword);
```

Comment:
my last comment from the conversation about hashing:

> the password in the database is stored hashed. Thus to compare them you must hash the password from the request first

Reply:
https://www.npmjs.com/package/bcrypt#to-check-a-password
check this example from bcrypt docs

Reply:
bcrypt.compare takes the plain text, and hashes it according to the hash algorithems that are visible in the stored hashed password
`$2a$10$qRmEayvDH5zdQd0sEeqccOzmpQb4s6gd2.zjQ0kul7JM8TWfXJQKO`

Reply:
I see, all good then!